proto/pulumi/language.proto

```diff
@@ -39,6 +39,9 @@ service LanguageRuntime {
 
     // GetProgramDependencies returns the set of dependencies required by the program.
     rpc GetProgramDependencies(GetProgramDependenciesRequest) returns (GetProgramDependenciesResponse) {}
+
+    // RunPlugin executes a plugin program and returns its result asynchronously.
+    rpc RunPlugin(RunPluginRequest) returns (stream RunPluginResponse) {}
```

Comment:
Frustrating that we can't have a tuple of immediate data and a trailing stream in a single RPC. The response API will look really funny if we want to add a message type for communicating a structured failure starting the plugin outside of the context of running it.